src/pushprocessor.ts

```diff
@@ -175,53 +268,17 @@ export class PushProcessor {
         if (!newRules.global.underride) newRules.global.underride = [];
 
         // Merge the client-level defaults with the ones from the server
-        const globalOverrides = newRules.global.override;
-        for (const originalOverride of DEFAULT_OVERRIDE_RULES) {
-            const existingRule = globalOverrides.find((r) => r.rule_id === originalOverride.rule_id);
-
-            // Dynamically add the user ID as the value for the is_user_mention rule.
-            let override: IPushRule;
-            if (originalOverride.rule_id === RuleId.IsUserMention) {
-                // If the user ID wasn't provided, skip the rule.
-                if (!userId) {
-                    continue;
-                }
-
-                override = JSON.parse(JSON.stringify(originalOverride)); // deep clone
-                override.conditions![0].value = userId;
-            } else {
-                override = originalOverride;
-            }
-
-            if (existingRule) {
-                // Copy over the actions, default, and conditions. Don't touch the user's preference.
-                existingRule.default = override.default;
-                existingRule.conditions = override.conditions;
-                existingRule.actions = override.actions;
```

Comment:
I fear we still need all this shit.

Reply:
The IsUserMention stuff is not needed because we don't have a default IsUserMention rule in the map, so we don't need to update its user id condition

The copying over actions etc is questionable, as we can't determine whether we or the server have the latest and greatest conditions/actions for a given rule

Reply:
> The IsUserMention stuff is not needed because we don't have a default IsUserMention rule in the map, so we don't need to update its user id condition

Huh. Good point. WTF was it doing there in the first place?

> The copying over actions etc is questionable, as we can't determine whether we or the server have the latest and greatest conditions/actions for a given rule

yeah, ok. I don't know if there are any instances of conditions/actions having changed on an existing specced rule, but it's certainly been a while and we'd need to have a careful think about it.